x/distribution/types/codec.go

```diff
@@ -37,6 +38,10 @@ func RegisterInterfaces(registry types.InterfaceRegistry) {
 		&MsgCommunityPoolSpend{},
 	)
 
+	registry.RegisterImplementations(
```

Comment:
We want to avoid this. Instead, if it's needed for testing. Register the type in the test.

Reply:
@amaurym advised that he thinks this is consensus breaking and was a bug in the first place.  I'm not totally familiar with the change though. Anything to say about this @amaurym ?

Reply:
I think we should keep this registration in v0.47 yeah. Imagine during a v046->v047 upgrade, there's an community spend proposal in state. Then, without this line, in the upgraded chain we wouldn't be able to decode that proposal, which is a bug (and would be consensus-breaking to fix).

An alternative is to add a migration, to update all old Content-based community spend proposals to use the new Msg-based one.

Either solution is good, but I advised to do it in a separate PR. But since this solution is already implemented here, I'm okay to continue this route.